tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -1572,6 +1572,19 @@ static void comp_data_get(uint16_t server_addr, uint8_t page, struct net_buf_sim
 {
 	uint8_t page_rsp;
 
+	/* Set Network Transmit Count state on the device greater than on provisioner to increase
+	 * probability of reception responses.
+	 */
+	uint8_t net_transmit;
+	uint8_t status;
+	int err;
+
+	net_transmit = BT_MESH_TRANSMIT(3, 20);
+	err = bt_mesh_cfg_cli_net_transmit_set(0, server_addr, net_transmit, &status);
+	if (err || status != net_transmit) {
+		FAIL("Net transmit set failed (err %d, transmit %x)", err, status);
+	}
+
 	net_buf_simple_reset(comp);
 	ASSERT_OK(bt_mesh_cfg_cli_comp_data_get(0, server_addr, page, &page_rsp, comp));
 	ASSERT_EQUAL(page, page_rsp);
```

Comment:
Would just adding delay after this string solve the problem?
```
/* Let complete advertising of the transaction to prevent collisions. */
k_sleep(K_SECONDS(1));
```
I guess the main problem is collisions of advertisement previous and current transactions.
Why I do not like the current solution? You just shift a little bit of communication timeline. However, I think the issue might happen again when it is shifted again by another PR.

Reply:
Of course if you add delay then you do not need to configure net transit option.

Reply:
<img width="838" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/26378606/cda10a74-3a43-4fff-b7e4-d7a2d7dbfb28">

@alxelax I modified it as you requested, but the test still failed, so I will have to go back to the previous version.